allennlp/data/token_indexers/token_characters_indexer.py

```diff
@@ -26,7 +26,11 @@ class TokenCharactersIndexer(TokenIndexer[List[int]]):
         We use a :class:`CharacterTokenizer` to handle splitting tokens into characters, as it has
         options for byte encoding and other things.  The default here is to instantiate a
         ``CharacterTokenizer`` with its default parameters, which uses unicode characters and
-        retains casing.
+        retains casing.  Note: this argument is depreciated in favor of tokenizer and will be
+        removed before version 1.0.
```

Comment:
@matt-peters, @joelgrus Can we just leave this indefinitely without an explicit deprecation warning? I'm not sure the churn of deleting this is worth it. If we _really_ want to delete it, let's provide a guarantee like "We will remove this argument _not_ before version i.j and YYYY/MM." That'll be more useful to users.

Reply:
(That will resolve the point @matt-gardner raised w.r.t. warnings as well.)